pkg/datapath/loader/wrapper/interface.go

```diff
@@ -0,0 +1,41 @@
+package wrapper
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/datapath"
+	"github.com/cilium/cilium/pkg/datapath/loader"
+	"github.com/cilium/cilium/pkg/datapath/loader/config"
+	"github.com/cilium/cilium/pkg/datapath/loader/metrics"
+)
+
+type LoaderWrapper struct{}
```

Comment:
Is there a reason to introduce this wrapper type rather than just convert the existing loader into this form with a singleton-style object like this that has methods?

Reply:
I substantially changed this in the most recent upload. Let me know what you think. A `Datapath` now has a `Loader`. We get this `Loader` in the endpoint by accessing the `Datapath` of the endpoint's `owner`.